server/service/src/goods_received/create_goods_received_shipment/mod.rs

```diff
@@ -0,0 +1,89 @@
+use repository::{
+    ActivityLogType, EqualFilter, Invoice, InvoiceFilter, InvoiceLineRowRepository,
+    InvoiceRepository, InvoiceRowRepository, RepositoryError,
+};
+
+use crate::{activity_log::activity_log_entry, service_provider::ServiceContext};
+
+mod generate;
+mod test;
+mod validate;
+
+use generate::*;
+use validate::*;
+
+#[derive(Debug, PartialEq)]
+pub struct CreateGoodsReceivedShipment {
+    pub goods_received_id: String,
+}
+
+#[derive(Debug, PartialEq)]
+
+pub enum CreateGoodsReceivedShipmentError {
+    DatabaseError(RepositoryError),
+    CreatedInvoiceDoesNotExist,
+    ProblemGettingOtherParty,
+    GoodsReceivedDoesNotExist,
+    PurchaseOrderDoesNotExist,
+    GoodsReceivedEmpty,
+    PurchaseOrderLinesNotFound(Vec<String>),
+    NotThisStoreGoodsReceived,
+    NotThisStorePurchaseOrder,
+    GoodsReceivedNotFinalised,
+    PurchaseOrderNotFinalised,
+}
+
+type OutError = CreateGoodsReceivedShipmentError;
+
+pub fn create_goods_received_shipment(
+    ctx: &ServiceContext,
+    input: CreateGoodsReceivedShipment,
+) -> Result<Invoice, OutError> {
+    let invoice = ctx
+        .connection
+        .transaction_sync(|connection| {
+            // TODO: Replace the following arguments with actual values as needed
```

Comment:
This should call `insert_inbound_shipment` through the service layer.
We'll need to call `insert_stock_in_line` for all the lines as well.

If we don't use the service layer, we're A) duplicating code and B) risking missing things in this code that are already handled in the service layer for inbound shipments